### PR TITLE
1210 silent migrate more options

### LINKED ISSUE #1210: bring back update_to_buf_dir
**Is your feature request related to a problem? Please describe.**
update_to_buf_dir option is unavailable

**Describe the solution you'd like**
update_to_buf_dir option is back

**Additional context**
Seems to have been removed via this commit: https://github.com/kyazdani42/nvim-tree.lua/commit/3ac443207f21bed29a2a3c2402f64098842ab753

image

### PR DESCRIPTION
**Pull request by alex-courtis**
closes #1210

### CODE REVIEW

File: lua/nvim-tree/legacy.lua [hunk 1]
```diff
@@ -190,6 +190,27 @@ local function refactored(opts)
       end
     end
   end
+
```
Comment (alex-courtis) [inline lua/nvim-tree/legacy.lua:193]: These are ugly.

I started to write a generic function to do this, but it was more ugly.

Comment (kyazdani42) [reply lua/nvim-tree/legacy.lua:193]: could you refactor the 3 line nested if statement into one in `refactored` ? Doesn't look right :smile:

File: lua/nvim-tree/legacy.lua [hunk 2]
```diff
@@ -190,6 +190,27 @@ local function refactored(opts)
...
+  if opts.view then
+    if opts.view.auto_resize ~= nil then
+      utils.table_create_missing(opts, "actions.open_file")
+      if opts.actions.open_file.resize_window == nil then
+        opts.actions.open_file.resize_window = opts.view.auto_resize
+      end
+      opts.view.auto_resize = nil
+    end
+  end
```
Comment (kyazdani42) [inline lua/nvim-tree/legacy.lua:213]: i think we can remove this though. A better strategy for the plugin stability is needed. I'll think about this and propose a solution that could work :)

File: lua/nvim-tree/legacy.lua (hunk 1 above)
Comment (alex-courtis) [reply lua/nvim-tree/legacy.lua:190]: Collapsed all.

File: lua/nvim-tree/legacy.lua (hunk 2 above)
Comment (alex-courtis) [reply lua/nvim-tree/legacy.lua]: Yes. This is very hacky.

However... we could just remove most of this code in a few months, as users will have migrated.

**Review by kyazdani42 — APPROVED** [review]

**Pull request merged**